include/data.h

```diff
@@ -102,6 +102,22 @@ struct TrainerClass
 
 #define TRAINER_ENCOUNTER_MUSIC(trainer)((gTrainers[trainer].encounterMusic_gender & 0x7F))
 
+struct Type
+{
+    u8 name[TYPE_NAME_LENGTH + 1];
+    u8 generic[17];
+    u8 palette;
+    u16 zMove;
+    u16 maxMove;
+    u16 enhanceItem;
+    u16 berry;
+    u16 gem;
+    u16 plate;
+    u16 memory;
+    u16 zCrystal;
+    u16 teraShard;
```

Comment:
Hmm, while useful as a reference for items associated to new types, the item fields aren't being used anywere in the code. Maybe adding them as comments as a reminder would be better if you're not planning in adding changes to the battle code.

Reply:
If this PR doesn't make any changes to the battle code (which I'm fine with) can that be captured in a later issue?

Reply:
I don't feel confortable introducing fields and data that aren't used at all :/

Reply:
I have to agree that if we add this information we may as well be using it in the same PR

Reply:
I thought the idea was that some of these fields exist mostly to facilitate uses like [psf's](<https://github.com/rh-hideout/pokeemerald-expansion/pull/4185#issuecomment-1943917623>)? Are there any official features which would want to know, e.g. "which Gem is associated with an arbitrary type?"

Reply:
largely agree with @mrgriffin

> I don't feel confortable introducing fields and data that aren't used at all :/

This has never been a principle we've had in expansion
* There are 500 different Pikachu forms that are straight up not being used, we can poll that and I'll bet $100 on that lmao
* we JUST merged ghoul's new field effect work that doesn't get used in the base game until the developer adds something
* I just merged in a new function for printing mon's height that doesn't get used anywhere

Like if we want to introduce that principle, I think its a different discussion entirely, but I don't want to hold this PR to a standard that doesn't actually exist yet

> Plus, there's also the fact that you'd be moving data away from gItemsInfo, where most item info is stored.

`gItemsInfo` doesn't appear in the PR at all, nothing is being moved out of there (unless I'm missing something).

I don't think its unreasonable or unusual to
- introduce a QoL improvement in PR 1
- iterate and improve on the use of that improvement in future PRs

_I am trying not to unnecessarily bloat the scope of this PR_.

that being said, I **think everybody is happy** if

- the fields in question get commented out for now
- this PR gets merged
- a future PR uncomments them and makes changes to the battle code to use them

Reply:
> * There are 500 different Pikachu forms that are straight up not being used, we can poll that and I'll bet $100 on that lmao

There's a difference between having Species data ready to use for the user for PoryMap and scripts and having data available for edge cases that need to write a custom function to use that data.

> * we JUST merged ghoul's new field effect work that doesn't get used in the base game until the developer adds something

Same with this, the new function is used in scripts to set up pre-battle conditions without the need to write custom code for it.

> * I just merged in a new function for printing mon's height that doesn't get used anywhere

Which function? Are you talking about `ReturnHeightStringNoWhitespace`? Because that's certainly being used by the Lotad/Seedot house.

> `gItemsInfo` doesn't appear in the PR at all, nothing is being moved out of there (unless I'm missing something).

I meant in context that if you were to use this data, you'd have to remove it from `gItemsInfo` to avoid reduncancy.

> * the fields in question get commented out for now
> * this PR gets merged
> * a future PR uncomments them and makes changes to the battle code to use them

Sounds good to me at least. Again, this info is useful as a reference, for adding new types, but it also needs a reason to justify adding unused data into the rom.

Reply:
> I meant in context that if you were to use this data, you'd have to remove it from `gItemsInfo` to avoid reduncancy.

I think both have value, depending on whether you're going from type to item, or item to type. (That's why earlier I suggested writing some quick tests that verify that if a type maps to something, the thing it maps to maps back to that type)